testing/upload_validation.ts

```diff
@@ -485,7 +494,10 @@ const variousSupportedAuthenticationValidators = Object.entries(defaultAuthentic
 const primitiveUnion = z.union([z.number(), z.string(), z.boolean(), z.date()]);
 
 const paramDefValidator = zodCompleteObject<ParamDef<any>>({
-  name: z.string().max(Limits.BuildingBlockName),
+  name: z
+    .string()
+    .max(Limits.BuildingBlockName)
+    .regex(regexFormulaName, 'Parameter names can only contain alphanumeric characters and underscores.'),
```

Comment:
I guess if we're trying to make sure the tokenizer doesn't choke on these, should use the exact same regex that the tokenizer users:

https://github.com/coda/coda/blob/main/modules/common/formula/tokens.ts#L248

(The above is copied from this file anyway, it was just the subset of the values relevant to formula names.)

Reply:
The only difference I see is that this upload_validation code doesn't allow an underscore as the first character. Am I misreading this? Is there still a tightening needed?

Reply:
Ah, are you saying we're stricter than the tokenizer is here, so it should be fine?

If so I agree but I think we should be more explicit about it. Could maybe assign `const regexParameterName = regexParameterName` and leave a note that this is currently the same as the tokenizer's restrictions except stricter because we don't allow leading underscores.

It seems like we're going to have bugs similar to the one you found if parameter and formula name validation ever diverges from what the tokenizer allows, so best to be really careful and explicit to minimize the likelihood of that happening again.

Reply:
Ok, done!